zipline/pipeline/factors/factor.py

```diff
@@ -33,17 +28,67 @@
     NumericalExpression,
     NUMEXPR_MATH_FUNCS,
     UNARY_OPS,
+    unary_op_name,
 )
 from zipline.pipeline.filters import (
     NumExprFilter,
     PercentileFilter,
 )
 from zipline.utils.control_flow import nullctx
+from zipline.utils.numpy_utils import (
+    bool_dtype,
+    datetime64ns_dtype,
+    float64_dtype,
+)
+from zipline.utils.preprocess import preprocess
 
 
 _RANK_METHODS = frozenset(['average', 'min', 'max', 'dense', 'ordinal'])
 
 
+def numbers_to_float64(func, argname, argvalue):
+    """
+    Preprocessor for converting numerical inputs into floats.
+
+    This is used in the binary operator constructors for Factor so that
+    `2 + Factor()` has the same behavior as `2.0 + Factor()`.
+    """
+    if isinstance(argvalue, Number):
+        return float64(argvalue)
```

Comment:
If numpy doesn't do this, then I'm not sure we should be in the business of doing it either.